TestFacetGrid.test_set_ticklabels failing with matplotlib 3.1.0

Running seaborn's test suite with matplotlib 3.1.0 (built from source) turns up the following failure from `test_axisgrid.py`:
```
=================================== FAILURES ===================================
______________________ TestFacetGrid.test_set_ticklabels _______________________

self = <seaborn.tests.test_axisgrid.TestFacetGrid object at 0x7f27afc6cf98>

    def test_set_ticklabels(self):
    
        g = ag.FacetGrid(self.df, row="a", col="b")
        g.map(plt.plot, "x", "y")
        xlab = [l.get_text() + "h" for l in g.axes[1, 0].get_xticklabels()]
        ylab = [l.get_text() for l in g.axes[1, 0].get_yticklabels()]
    
        g.set_xticklabels(xlab)
        g.set_yticklabels(rotation=90)
    
        got_x = [l.get_text() for l in g.axes[1, 1].get_xticklabels()]
        got_y = [l.get_text() for l in g.axes[0, 0].get_yticklabels()]
        npt.assert_array_equal(got_x, xlab)
        npt.assert_array_equal(got_y, ylab)
    
        x, y = np.arange(10), np.arange(10)
        df = pd.DataFrame(np.c_[x, y], columns=["x", "y"])
        g = ag.FacetGrid(df).map(pointplot, "x", "y", order=x)
        g.set_xticklabels(step=2)
        got_x = [int(l.get_text()) for l in g.axes[0, 0].get_xticklabels()]
        npt.assert_array_equal(x[::2], got_x)
    
        g = ag.FacetGrid(self.df, col="d", col_wrap=5)
        g.map(plt.plot, "x", "y")
        g.set_xticklabels(rotation=45)
        g.set_yticklabels(rotation=75)
        for ax in g._bottom_axes:
            for l in ax.get_xticklabels():
>               nt.assert_equal(l.get_rotation(), 45)

tests/test_axisgrid.py:485: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
<embedded stdlib>/unittest/case.py:829: in assertEqual
    assertion_func(first, second, msg=msg)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <nose.tools.trivial.Dummy testMethod=nop>, first = 0.0, second = 45
msg = '0.0 != 45'

    def _baseAssertEqual(self, first, second, msg=None):
        """The default assertEqual implementation, not type specific."""
        if not first == second:
            standardMsg = '%s != %s' % _common_shorten_repr(first, second)
            msg = self._formatMessage(msg, standardMsg)
>           raise self.failureException(msg)
E           AssertionError: 0.0 != 45

<embedded stdlib>/unittest/case.py:822: AssertionError
```

This looks like it could be related to https://github.com/mwaskom/seaborn/issues/1598, though I can't say for sure.

--- a/seaborn/axisgrid.py
+++ b/seaborn/axisgrid.py
@@ -892,7 +892,7 @@ class FacetGrid(Grid):
         return self
 
     def set_xticklabels(self, labels=None, step=None, **kwargs):
-        """Set x axis tick labels of the grid."""      
+        """Set x axis tick labels of the grid."""
         for ax in self.axes.flat:
             if labels is None:
                 curr_labels = [l.get_text() for l in ax.get_xticklabels()]

--- a/seaborn/axisgrid.py
+++ b/seaborn/axisgrid.py
@@ -892,23 +892,27 @@ class FacetGrid(Grid):
         return self
 
     def set_xticklabels(self, labels=None, step=None, **kwargs):
-        """Set x axis tick labels on the bottom row of the grid."""
+        """Set x axis tick labels of the grid."""      
         for ax in self.axes.flat:
             if labels is None:
-                labels = [l.get_text() for l in ax.get_xticklabels()]
+                curr_labels = [l.get_text() for l in ax.get_xticklabels()]
                 if step is not None:
                     xticks = ax.get_xticks()[::step]
-                    labels = labels[::step]
+                    curr_labels = curr_labels[::step]
                     ax.set_xticks(xticks)
-            ax.set_xticklabels(labels, **kwargs)
+                ax.set_xticklabels(curr_labels, **kwargs)
+            else:
+                ax.set_xticklabels(labels, **kwargs)
         return self
 
     def set_yticklabels(self, labels=None, **kwargs):
         """Set y axis tick labels on the left column of the grid."""
         for ax in self.axes.flat:
             if labels is None:
-                labels = [l.get_text() for l in ax.get_yticklabels()]
-            ax.set_yticklabels(labels, **kwargs)
+                curr_labels = [l.get_text() for l in ax.get_yticklabels()]
+                ax.set_yticklabels(curr_labels, **kwargs)
+            else:
+                ax.set_yticklabels(labels, **kwargs)
         return self
 
     def set_titles(self, template=None, row_template=None,  col_template=None,

